### PR TITLE
A first re-org pass

### PR DESCRIPTION
**Pull request by jmlarson1**
In response to issue #850, thoughts on these high-level categories here? Organize alphabetical in the lower categories?

### CODE REVIEW

**Comment by jlnav** [comment]
Since nothing here is literal-included, could the sub-categories be divided into rst sub-headings so it's easier to see them on a glance? Thoughts?

**Comment by jmlarson1** [comment]
I'm not sure this is possible. When I tried `**header**` or 

````
header
------
````
they rendered with the "`-`" and "`*`" (not formatted).  Can you show me what you are thinking?

**Comment by shuds13** [comment]
I think the categories are fine. Maybe add a 'general' sub-heading for those that dont fit in other categories. And maybe add a line before the schedular/resource_info section to say these are for sub-dictionaries to above options.

**Comment by jmlarson1** [comment]
@shuds13 I've implemented your suggestions. I'm not sure why the first block of text (starting line 8) is formatted differently than the second block of text (starting line 170). Not sure it matters right now.

File: docs/data_structures/libE_specs.rst [hunk 1]
```diff
@@ -128,12 +146,14 @@ Specifications for libEnsemble::
         'resource_info' [dict]:
             Provide resource information that will override automatically detected resources.
             The allowable fields are given below in 'Overriding Auto-detection'
-            Note that if ``disable_resource_manager`` is set then
+            Note that if "disable_resource_manager" is set then
             this option is ignored.
         'scheduler_opts' [dict]:
             A dictionary of options for the resource scheduler.
             See 'Scheduler Options' for more.
```
Comment (wildsm) [inline docs/data_structures/libE_specs.rst:153]: See 'Scheduler Options' for more options.

(add " options")

File: docs/data_structures/libE_specs.rst [hunk 2]
```diff
@@ -144,32 +164,32 @@ See options for :ref:`built-in scheduler<resources-scheduler>`.
 Overriding Auto-detection
```
Comment (wildsm) [inline docs/data_structures/libE_specs.rst:164]: Overriding Auto-Detection

File: docs/data_structures/libE_specs.rst [hunk 3]
```diff
@@ -24,6 +28,19 @@ Specifications for libEnsemble::
             Whether libEnsemble should save the states of H and persis_info on
             aborting after an error.
             Default: True
+        'worker_timeout' [int]:
+            When libEnsemble concludes and attempts to close down worker processes,
+            the number of seconds after which workers are considered timed out. Worker
+            processes are then terminated. multiprocessing default: 1
```
Comment (jmlarson1) [inline docs/data_structures/libE_specs.rst:34]: Is this `1 second` or something else?

**Comment by jlnav** [comment]
> Can we add "only" to the `nworkers` statement about tcp/local modes?
> 
> Also, can we organize within each grouping?

What subcategories are you considering?

**Comment by jmlarson1** [comment]
> > Can we add "only" to the `nworkers` statement about tcp/local modes?
> > Also, can we organize within each grouping?
> 
> What subcategories are you considering?

I was thinking (for example) to have `Directory management options:` be alphabetical in order (or maybe `use_worker_dir` to be first.

**Comment by shuds13** [comment]
It seems the line

The following describe the dictionary options within ``libE_specs``.

was added but then removed?

**Comment by jmlarson1** [comment]
I think it was inadvertently dropped in ceba837e 

I added it back

**Pull request merged**